src/jit/compiler.cpp

```diff
@@ -2443,6 +2443,74 @@ const char* Compiler::compLocalVarName(unsigned varNum, unsigned offs)
 #endif // DEBUG
 /*****************************************************************************/
 
+#ifdef _TARGET_XARCH_
+static bool configEnableISA(InstructionSet isa)
+{
+#ifdef DEBUG
+    switch (isa)
+    {
+        case InstructionSet_SSE:
+            return JitConfig.EnableSSE() != 0;
+        case InstructionSet_SSE2:
+            return JitConfig.EnableSSE2() != 0;
+        case InstructionSet_SSE3:
+            return JitConfig.EnableSSE3() != 0;
+        case InstructionSet_SSSE3:
+            return JitConfig.EnableSSSE3() != 0;
+        case InstructionSet_SSE41:
+            return JitConfig.EnableSSE41() != 0;
+        case InstructionSet_SSE42:
+            return JitConfig.EnableSSE42() != 0;
+        case InstructionSet_AVX:
+            return JitConfig.EnableAVX() != 0;
+        case InstructionSet_AVX2:
+            return JitConfig.EnableAVX2() != 0;
+
+        case InstructionSet_AES:
+            return JitConfig.EnableAES() != 0;
+        case InstructionSet_BMI1:
+            return JitConfig.EnableBMI1() != 0;
+        case InstructionSet_BMI2:
+            return JitConfig.EnableBMI2() != 0;
+        case InstructionSet_FMA:
+            return JitConfig.EnableFMA() != 0;
+        case InstructionSet_LZCNT:
+            return JitConfig.EnableLZCNT() != 0;
+        case InstructionSet_PCLMULQDQ:
+            return JitConfig.EnablePCLMULQDQ() != 0;
+        case InstructionSet_POPCNT:
+            return JitConfig.EnablePOPCNT() != 0;
+        default:
+            return false;
+    }
+#else
+    switch (isa)
+    {
+        case InstructionSet_SSE:
+        case InstructionSet_SSE2:
+        case InstructionSet_SSE3:
+        case InstructionSet_SSSE3:
+        case InstructionSet_SSE41:
+        case InstructionSet_SSE42:
+        case InstructionSet_AVX:
+        case InstructionSet_AVX2:
+            return (JitConfig.JITHWIntrinsicISA1() & (1ULL << isa)) != 0;
+
+        case InstructionSet_AES:
+        case InstructionSet_BMI1:
+        case InstructionSet_BMI2:
+        case InstructionSet_FMA:
+        case InstructionSet_LZCNT:
+        case InstructionSet_PCLMULQDQ:
+        case InstructionSet_POPCNT:
+            return (JitConfig.JITHWIntrinsicISA2() & (1ULL << (isa - 32))) != 0;
+        default:
+            return false;
+    }
```

Comment:
Why do you need to introduce JITHWIntrinsicISA1/2? Why don't you just make all the EnableXXX variables available in non-DEBUG builds?

Reply:
This is from @CarolEidt ‘s suggestion https://github.com/dotnet/coreclr/pull/14456#issuecomment-337012132

Reply:
I don't think it's necessary or desirable to make them all available in non-DEBUG builds, as I don't think we want a large proliferation of Release mode flags. That said, I'm not sure how much value there is in exposing them at all (even as a mask) in Release. Since control of these is primarily desirable from a test perspective, I think we can do that testing in Checked/Debug. I apologize if my comment led to confusion.
I would be in favor of having *only* the DEBUG mode flags, for testing purposes, and omitting the `JITHWIntrinsicISA1` and `JITHSIntrinsicISA2`.